tlatools/org.lamport.tlatools/src/tlc2/tool/fp/LongArray.java

```diff
@@ -43,10 +43,10 @@
  * In 2012 this poses a too hard limit on the usable memory, hence we trade
  * generality for performance.
  */
-@SuppressWarnings("restriction")
 public final class LongArray {
 
-	private final Unsafe unsafe;
+	@SuppressWarnings("restriction")
+	private final sun.misc.Unsafe unsafe;
```

Comment:
Do the changes in this file do anything? (If not, let's revert them to keep the diff clean.)

Reply:
They kind of do. Moving an import to a fully qualified name is the recommended way to suppress warnings. And then attempting to suppress the warning on a specific instance of the warning instead of the entire class shows it doesn't work for some reason. Could go either way.

Reply:
Okay I somehow figured out how to silence these warnings (by blanket-silencing the entire file at the compiler level). You used to be able to use `-XDenableSunApiLintControl -Xlint:-sunapi` along with `@SuppressWarnings("sunapi")` for this specific case but that was removed in Java 9.

Reply:
https://github.com/tlaplus/tlaplus/blob/fa4968f60a0a460da4b7e182330680ce93bd96aa/tlatools/org.lamport.tlatools/customBuild.xml#L142-L149

Reply:
Yeah the unsafe warnings aren't emitted anymore!